### ISSUE TITLE
[Feature Req] - Add "Save Repository" Functionality for Easier Access

### ISSUE THREAD

**Issue by imkeanserna** [MohamedDanis/opnhub#35]
Description:
I would like to suggest a feature that allows users to "save" repositories directly from the site. Since I understand that there is currently no user authentication or account system, this feature can be implemented using the browser's localStorage. This would enable each user to store their saved repositories locally in their browser, allowing for quick access to bookmarked repositories without the need for an external database or login

![image](https://github.com/user-attachments/assets/f365b804-b741-4c3f-bb46-7667e790fb40)

**Comment by imkeanserna** [comment]
@MohamedDanis can i implement this feature?

**Comment by MohamedDanis** [comment]
that's great, and we need another page for showing saved respositories , right ?

**Comment by imkeanserna** [comment]
Yes, we need another page for saved repositories

**Comment by MohamedDanis** [comment]
I thought about adding authentication and database to this app, what do you think ?

**Comment by imkeanserna** [comment]
Yes, I think it would be a great idea! If users want to save repositories, they should be required to log in first. We could authenticate them using their GitHub account, as it makes more sense in this context compared to Gmail.

**Comment by MohamedDanis** [comment]
Despite the server costs, I am considering using Supabase. What are your thoughts?

**Comment by imkeanserna** [comment]
I tried Supabase before, but it felt slow to me, which is why I switched to Aiven. It's much faster, especially when migrating databases.

**Comment by MohamedDanis** [comment]
is it free ?

**Comment by imkeanserna** [comment]
Yes, there is a free plan https://aiven.io/docs/platform/concepts/free-plan

**Comment by MohamedDanis** [comment]
I will check and update it here

**Comment by MohamedDanis** [comment]
@imkurosaki  You can create a page for saved repositories using dummy data and make a pull request. I can handle the authentication using NextAuth as planned. Do you have any specific preferences for the authentication?

**Comment by imkeanserna** [comment]
Hey @MohamedDanis I’m currently in my final year and things are a bit hectic, so I won’t be able to work on this feature right now. Could we connect on LinkedIn instead? Yes, nextauth is very customizable and we can do clerk if you prefer a user has OTP verification to their email if you want to add email authentication and they can do github confirmation also.

**Comment by MohamedDanis** [comment]
no worries, I will work on it

**Comment by MohamedDanis** [comment]
open for contributions
